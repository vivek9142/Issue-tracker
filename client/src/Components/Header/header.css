.nav-items-container{
    display: flex;
    justify-content: space-between;
}
img.nav--icon {
    width: 24px;
    margin-right: 10px;
}

@media screen and (max-width: 992px) {
    .nav-items-container{
        display: block;
    }
    img.nav--icon{
        width: 20px;
        margin-right: 8px;
    }
    .auth{
        margin: .8rem 0;
    }
    .auth .btn{
        padding: .1rem .5rem;
    }
  }