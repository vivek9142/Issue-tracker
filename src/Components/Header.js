import React from 'react';
import {Link } from 'react-router-dom';



import { useDispatch,useSelector } from 'react-redux';
import { userActions } from '../Redux/Reducer/userReducer';

const Header = (props) => {
  
  const  dispatch = useDispatch();
  const currentUser = useSelector(state => state.user.authenticatedUser);

  const logoutHandler = () => {
      dispatch(userActions.logoutUser());
  }
    return (      
        <>
        <Link to='/About'>About</Link>
        <Link to='/'>Issues</Link>
        { Object.keys(currentUser).length === 0 &&
            <React.Fragment>
            <Link to='/Register'>Register</Link>
            <Link to='/Login'>Login</Link>
            </React.Fragment> 
        }
        
        

        { Object.keys(currentUser).length !== 0 &&
            <React.Fragment>
            <Link to='/AddIssue'>Add Issue</Link>
            <span>Logged in as {`${currentUser.firstName} ${currentUser.lastName}`}</span>
            <button onClick={logoutHandler}>Logout</button>
            </React.Fragment>
        }
        </>
    );
}

export default Header;