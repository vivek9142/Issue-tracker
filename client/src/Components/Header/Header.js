import React from 'react';
import {Link } from 'react-router-dom';
import { useDispatch,useSelector } from 'react-redux';
import { userActions } from '../../Redux/Reducer/userReducer';

import loginImg from '../../Assets/login.png';
import registerImg from '../../Assets/register.png';
import logoutImg from '../../Assets/logout.png';

import './header.css';

const Header = (props) => {
  
  const  dispatch = useDispatch();
  const currentUser = useSelector(state => state.user.authenticatedUser);

  const logoutHandler = () => {
      dispatch(userActions.logoutUser());
  }
    return (  
        <>
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark navbar-sticky">
        <div className="container">
          <Link className="navbar-brand" to="/">Issue Tracker</Link>
          <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
            <span className="navbar-toggler-icon"></span>
          </button>
          <div className="collapse navbar-collapse nav-items-container" id="navbarNavAltMarkup">
            <div className="navbar-nav">
              <Link className="nav-link" to="/About">About</Link>
              <Link className="nav-link" to="/Top-Viewed-Issues">Top Viewed Issues</Link>
            </div>
            
            <div className="auth d-flex">
              { Object.keys(currentUser).length === 0 &&
              <React.Fragment>
                  <Link className="btn btn-primary me-2" to="/Register">
                    <img className='nav--icon' src={registerImg} alt="register-icon" />
                    Register</Link>
                  <Link className="btn btn-success" to="/Login">
                      <img className='nav--icon' src={loginImg} alt="login-icon" />
                      Login</Link>
              </React.Fragment> 
              }

              { Object.keys(currentUser).length !== 0 &&
              <React.Fragment>
                  <span className='navbar-text me-5'>Logged in as {`${currentUser.firstName} ${currentUser.lastName}`}</span>
                  
                  <button className="btn btn-success" to="/Logout" onClick={logoutHandler}>
                      <img className='nav--icon' src={logoutImg} alt="logout-icon" />
                      Logout</button>
              </React.Fragment> 
              }
            </div>
          </div>
        </div>
      </nav>

        </>
    );
}

export default Header;