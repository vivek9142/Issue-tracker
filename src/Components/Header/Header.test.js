import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from './Header';
import {Provider} from 'react-redux';
import store from '../../Redux/Store';
import {MemoryRouter} from 'react-router-dom';

describe('Header component Page Test Suite', () => {
    beforeEach(()=>{
    render(<MemoryRouter>
        <Provider store={store}><Header/></Provider>
        </MemoryRouter>);
    });
    test('Header component Snapshot',()=>{
        const tree = renderer.create(   
        <MemoryRouter>
        <Provider store={store}><Header/></Provider>
        </MemoryRouter>
            );
        expect(tree).toMatchSnapshot();
      });
      
    test('renders main Brand logo Heading of Header component Page', () => {
        const element = screen.getByText(/Issue Tracker/i);
        expect(element).toBeInTheDocument();
    });

    test('Header component should have About and Top Viewed Issues Links', () => {
        const element1 = screen.getByText(/About/i);
        expect(element1).toBeInTheDocument();
        const element2 = screen.getByText(/Top Viewed Issues/i);
        expect(element2).toBeInTheDocument();
    });

    test('Header component should have Register and Login Links', () => {
        const element1 = screen.getByText(/Register/i);
        expect(element1).toBeInTheDocument();
        const element2 = screen.getByText(/Login/i);
        expect(element2).toBeInTheDocument();
    });

  
})