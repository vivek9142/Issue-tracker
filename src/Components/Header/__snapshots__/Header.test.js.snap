// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Page Test Suite Header component Snapshot 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark navbar-sticky"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Issue Tracker
    </a>
    <button
      aria-controls="navbarNavAltMarkup"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarNavAltMarkup"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse nav-items-container"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-link"
          href="/About"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="nav-link"
          href="/Top-Viewed-Issues"
          onClick={[Function]}
        >
          Top Viewed Issues
        </a>
      </div>
      <div
        className="auth d-flex"
      >
        <a
          className="btn btn-primary me-2"
          href="/Register"
          onClick={[Function]}
        >
          <img
            alt="register-icon"
            className="nav--icon"
            src="register.png"
          />
          Register
        </a>
        <a
          className="btn btn-success"
          href="/Login"
          onClick={[Function]}
        >
          <img
            alt="login-icon"
            className="nav--icon"
            src="login.png"
          />
          Login
        </a>
      </div>
    </div>
  </div>
</nav>
`;
